src/main/java/core/basesyntax/services/impl/ReportGeneratorServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.services.ReportGeneratorService;
+
+public class ReportGeneratorServiceImpl implements ReportGeneratorService {
+    private final FruitDao fruitDao;
+
+    public ReportGeneratorServiceImpl(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public String getReport() {
+        StringBuilder report = new StringBuilder();
+        report.append("fruit,quantity").append(System.lineSeparator());
+        fruitDao.getStorage().forEach((fruit, quantity) -> report.append(fruit)
+                .append(",")
+                .append(quantity)
+                .append(System.lineSeparator()));
+
+        return report.toString();
+    }
```

Comment:
It's good practice to follow the encapsulation principle. The storage map should not be exposed directly. Instead, the FruitDao should provide methods to interact with the storage without exposing its internal representation.